feat: projects with permission in a single query

dogukankaratas opened the pull request:
Fetching all the projects once and checking their permission one-by-one was a quite expensive operation, instead of that I implemented ProjectsWithPermissions class in specklepy and this PR adapts this functionality into our connector and make it x2 faster to initialize.

dogukankaratas committed:
optimized query

bimgeek commented on bpy_speckle/connector/utils/project_manager.py:
```diff
@@ -64,3 +77,80 @@ def get_projects_for_account(
+        include_implicit_access=True,
+    )
+
+    projects = client.active_user.get_projects(limit=50, filter=filter).items
```
why set projects limit to 50? Isn't that a lot?

bimgeek commented on bpy_speckle/connector/utils/project_manager.py:
```diff
@@ -64,3 +77,80 @@ def get_projects_for_account(
+        include_implicit_access=True,
+    )
+
+    projects = client.active_user.get_projects(limit=50, filter=filter).items
```
same as above.

bimgeek commented on bpy_speckle/connector/utils/project_manager.py:
```diff
@@ -25,37 +23,52 @@ def get_projects_for_account(
-                    can_load_permission,
+            projects_with_permissions = (
+                workspace_resource.get_projects_with_permissions(
+                    workspace_id=workspace_id, limit=10
```
inconsistent project limits. below is 50, here it's 10. Maybe we should have a DEFAULT_PROJECT_LIMIT = 10 and use that as the input?

bimgeek replied:
search filter is missing.

dogukankaratas committed:
search added

dogukankaratas committed:
filter updated

dogukankaratas committed:
optimization on workspace id

dogukankaratas committed:
adds personal projects

dogukankaratas committed:
Merge branch 'v3-dev' into dogukan/query-optimizations

bimgeek committed:
fail when trying to create a model in personal workspace

bimgeek committed:
redraw after selection

Ensures the UI updates immediately after adding an object
to the selection filter dialog. This provides more responsive feedback to the user.
